service/administration/userGroup.go

```diff
@@ -0,0 +1,305 @@
+package administration
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/spotinst/spotinst-sdk-go/spotinst"
+	"github.com/spotinst/spotinst-sdk-go/spotinst/client"
+	"github.com/spotinst/spotinst-sdk-go/spotinst/util/jsonutil"
+	"github.com/spotinst/spotinst-sdk-go/spotinst/util/uritemplates"
+)
+
+type UserGroup struct {
+	Description *string            `json:"description,omitempty"`
+	Name        *string            `json:"name,omitempty"`
+	Policies    []*UserGroupPolicy `json:"policies,omitempty"`
+	UserIds     []string           `json:"userIds,omitempty"`
+	UserGroupId *string            `json:"id,omitempty"`
+	Users       []*UserGroupUser   `json:"users,omitempty"`
+	CreatedAt   *string            `json:"createdAt,omitempty"`
+	PolicyNames []string           `json:"policyNames,omitempty"`
+	UsersCount  *int               `json:"usersCount,omitempty"`
+
+	// forceSendFields is a list of field names (e.g. "Keys") to
+	// unconditionally include in API requests. By default, fields with
+	// empty values are omitted from API requests. However, any non-pointer,
+	// non-interface field appearing in ForceSendFields will be sent to the
+	// server regardless of whether the field is empty or not. This may be
+	// used to include empty fields in Patch requests.
+	forceSendFields []string
+
+	// nullFields is a list of field names (e.g. "Keys") to include in API
+	// requests with the JSON null value. By default, fields with empty
+	// values are omitted from API requests. However, any field with an
+	// empty value appearing in NullFields will be sent to the server as
+	// null. It is an error if a field in this list has a non-empty value.
+	// This may be used to include null fields in Patch requests.
+	nullFields []string
+}
+
+type UserGroupPolicy struct {
+	AccountIds []string `json:"accountIds,omitempty"`
+	PolicyId   *string  `json:"policyId,omitempty"`
+	PolicyName *string  `json:"policyName,omitempty"`
+	PolicyType *string  `json:"policyType,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+type UserGroupUser struct {
+	Type     *string `json:"type,omitempty"`
+	UserId   *string `json:"userId,omitempty"`
+	UserName *string `json:"userName,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+type ListUserGroupsInput struct{}
+
+type ListUserGroupsOutput struct {
+	UserGroups []*UserGroup `json:"userGroups,omitempty"`
+}
+
+type CreateUserGroupOutput struct {
+	UserGroup *UserGroup `json:"userGroup,omitempty"`
+}
+
+type ReadUserGroupInput struct {
+	UserGroupID *string `json:"id,omitempty"`
+}
+
+type ReadUserGroupOutput struct {
+	UserGroup *UserGroup `json:"userGroup,omitempty"`
+}
+
+type UpdateUserGroupInput struct {
+	UserGroupID *string `json:"id,omitempty"`
+}
+
+type UpdateUserGroupOutput struct {
+	UserGroup *UserGroup `json:"userGroup,omitempty"`
```

Comment:
update user group details doesn't return anything, check once if it's really required

Reply:
removed it and changed update user grou pimplementation